shared/datastore.go

```diff
@@ -73,13 +71,16 @@ func LoadTestRunKeys(
 			query = query.Filter("TimeStart <", *to)
 		}
 
-		fetched, err := query.KeysOnly().GetAll(ctx, nil)
-		if err != nil {
-			return nil, err
-		}
 		var keys []*datastore.Key
-		for _, key := range fetched {
-			if limit == nil || *limit > len(keys) && len(keys) < MaxCountMaxValue {
+		iter := query.KeysOnly().Run(ctx)
+		for {
+			key, err := iter.Next(nil)
+			if err == datastore.Done {
+				break
+			} else if err != nil {
+				return result, err
+			}
+			if (limit == nil || *limit > len(keys)) && len(keys) < MaxCountMaxValue {
```

Comment:
We should add an `else` branch for this `if` and break the loop there to prevent unnecessary full table scan.

Reply:
Done (and other simplification).